### PR TITLE
refactor: atomize every form elements

### PR DESCRIPTION
**Pull request by rdubigny**

### CODE REVIEW

File: src/components/Form/components/TelInput.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import Helper from '../../Helper';
+
+export const TelInput = ({
+  label,
+  helper,
+  name,
+  value = null,
+  disabled,
+  onChange,
+  'aria-label': ariaLabel,
```
Comment (XavierJp) [inline src/components/Form/components/TelInput.js:12]: kebab case is unexpected in props

File: src/components/Form/components/TextAreaInput.js [hunk 2]
```diff
@@ -10,6 +10,7 @@ export const TextAreaInput = ({
   value = null,
   disabled,
   onChange,
+  rows = 10,
```
Comment (XavierJp) [inline src/components/Form/components/TextAreaInput.js:13]: noice

File: src/components/Form/index.js [hunk 3]
```diff
@@ -81,6 +81,7 @@ export const Form = ({
         update: true,
         send_application: true, // Enable edition for new enrollment (ie. enrollment has no id)
       },
+      demarche: 'default',
```
Comment (XavierJp) [inline src/components/Form/index.js:84]: always in default state ? Or only when demarches is defined ?

File: src/components/form-sections/CadreJuridiqueSection.js [hunk 4]
```diff
@@ -21,49 +42,28 @@ const CadreJuridiqueSection = ({ CadreJuridiqueDescription = () => null }) => {
...
+          </>
+        }
+        name="fondement_juridique_title"
+        placeholder="« loi », « décret », « délibération », etc."
+        value={fondement_juridique_title}
+        disabled={disabled}
+        onChange={onChange}
+        rows={1}
+      />
+      <OrWrapper>
```
Comment (XavierJp) [inline src/components/form-sections/CadreJuridiqueSection.js:59]: noice

File: src/components/form-sections/CguSection/FcHasAlternativeAuthenticationMethod.js [hunk 5]
```diff
@@ -1,29 +1,25 @@
...
-      htmlFor="has_alternative_authentication_methods"
-      className="label-inline"
-    >
-      J’atteste que mon service propose une alternative à la connexion avec
-      FranceConnect, et que cette alternative permet l’accès, dans des
-      conditions analogues, à la même prestation de service public.
-    </label>
-  </div>
+  <CheckboxInput
+    label={
```
Comment (XavierJp) [inline src/components/form-sections/CguSection/FcHasAlternativeAuthenticationMethod.js:11]: could be using `children`

File: src/components/form-sections/CguSection/index.js [hunk 6]
```diff
@@ -18,26 +19,22 @@ const Index = ({
...
-          <a href={cguLink} target="_blank" rel="noreferrer noopener">
-            conditions générales d'utilisation
-          </a>{' '}
-          et je les valide. Je confirme que le délégué à la protection des
-          données de mon organisation est informé de ma demande.
-        </label>
-      </div>
-
+      <CheckboxInput
+        label={
```
Comment (XavierJp) [inline src/components/form-sections/CguSection/index.js:23]: same

File: src/components/form-sections/DemarcheSection/DemarcheSectionSelect.js [hunk 7]
```diff
@@ -72,25 +73,23 @@ export const DemarcheSectionSelect = () => {
...
-          >
-            {Object.keys(demarches).map(demarcheId => (
-              <option key={demarcheId} value={demarcheId}>
-                {get(demarches, demarcheId, {}).label}
-              </option>
-            ))}
-          </select>
-        </div>
+        <Select
+          label={
```
Comment (XavierJp) [inline src/components/form-sections/DemarcheSection/DemarcheSectionSelect.js:77]: Or just a string

File: src/components/form-sections/PreviousEnrollmentSection/index.js [hunk 8]
```diff
@@ -70,17 +71,6 @@ const PreviousEnrollmentSection = ({
...
-      target: {
-        name: 'previous_enrollment_id',
-        value: id,
-      },
-    });
-  };
-
   return (
     <>
       {!disabled && !isUserEnrollmentLoading && (
```
Comment (XavierJp) [inline src/components/form-sections/PreviousEnrollmentSection/index.js:76]: this is repeated 4x time, maybe we could set it in a more meaningful variable name

File: src/components/form-sections/PreviousEnrollmentSection/index.js [hunk 9]
```diff
@@ -171,31 +160,31 @@ const PreviousEnrollmentSection = ({
...
+                <Select
+                  label={
+                    <>
+                      Nom de la démarche{' '}
+                      <b>{TARGET_API_LABELS[previousTargetApi]}</b>
+                    </>
+                  }
+                  helper={
+                    target_api === 'api_impot_particulier_fc_sandbox' &&
+                    'Sélectionnez "aucune démarche" si vous souhaitez accéder à l\'API sans FranceConnect'
```
Comment (XavierJp) [inline src/components/form-sections/PreviousEnrollmentSection/index.js:173]: you can use ’ instead of '

**Review by XavierJp — CHANGES_REQUESTED** [review]

File: src/components/Form/index.js (hunk 3 above)
Comment (rdubigny) [reply src/components/Form/index.js]: committed by error! nicely spotted Xav Eagle-eyed Jp!

File: src/components/form-sections/PreviousEnrollmentSection/index.js (hunk 8 above)
Comment (rdubigny) [reply src/components/form-sections/PreviousEnrollmentSection/index.js:76]: we will delete this component in the future

**Pull request merged**